warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/WhindexIngest.java

```diff
@@ -0,0 +1,212 @@
+package datawave.ingest.data.config.ingest;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.hadoop.conf.Configuration;
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.LinkedListMultimap;
+import com.google.common.collect.Multimap;
+
+import datawave.ingest.data.Type;
+import datawave.ingest.data.config.NormalizedContentInterface;
+import datawave.ingest.data.config.NormalizedFieldAndValue;
+import datawave.marking.MarkingFunctions;
+import datawave.marking.MarkingFunctionsFactory;
+
+public interface WhindexIngest {
```

Comment:
Can we include an overall comment which describes what a whindex is, and includes the overall configuration syntax please

Reply:
See pinned comment in out chat:

So the configuration needs to be able to encode the following:
1) The field from which the values will be pulled.
2) A mapping of fieldname + value -> whindex fieldname
The separator configuration is nice but perhaps not really needed.  The current configuration is
RULES := RULE;RULE
RULE := FIELDS::VALUE{,VALUE}
FIELDS := FIELD{,FIELD}
FIELD := SRC_FIELD:DST_FIELD[:DELETE_SRC]
DELETE_SRC := 'true' if the SCR_FIELD should be removed, 'false' otherwise - defaults to 'false'
SRC_FIELD: the field which should be renamed/remapped
DST_FIELD := the new name for the field (a.k.a the whindex field)
VALUE := The value pulled from the field defined in #1 above.

Reply:
Lets call the field defined in 1. above to be the VALUE_FIELD

Reply:
So this ingester needs to be able to create the following fields:

DST_FIELD = <value of SRC_FIELD> IFF VALUE_FIELD contains one of the VALUEs

And the ingester needs to remove the following fields:

SRC_FIELD IFF DELETE_SRC is true and its value is being used in the DST_FIELD created above.

Reply:
Just to confirm, is VALUE_FIELD equivalent to SRC_FIELD within the rule configuration syntax?

Reply:
If VALUE_FIELD is not equivalent to SRC_FIELD, then I assume we need another configuration property such as `<type>.whindex.valueField`

Reply:
VALUE_FIELD clarified in separate chat.